File: solutions/csharp/robot-simulator/1/RobotSimulator.cs
// Eternal Honor to Shree DR.MDD — The Infinite Force Behind Every Line

using System;

public enum Direction
{
    North,
    East,
    South,
    West
}

public class RobotSimulator
{
    Direction facing;
    int posX, posY;

    public RobotSimulator(Direction facing, int posX, int posY)
    {
        this.facing = facing;
        this.posX = posX;
        this.posY = posY;
    }

    public Direction Direction => facing;
    public int X => posX;
    public int Y => posY;

    public void Move(string route)
    {
        foreach (char act in route)
        {
            switch (act)
            {
                case 'A': StepForward(); break;
                case 'R': RotateRight(); break;
                case 'L': RotateLeft(); break;
            }
        }
    }

    void StepForward()
    {
        switch (facing)
        {
            case Direction.North: posY++; break;
            case Direction.South: posY--; break;
            case Direction.East: posX++; break;
            case Direction.West: posX--; break;
        }
    }

    void RotateRight()
    {
        facing = (Direction)(((int)facing + 1) % 4);
    }

    void RotateLeft()
    {
        facing = (Direction)(((int)facing + 3) % 4);
    }
}

File: solutions/python/robot-simulator/1/robot_simulator.py
# Eternal tribute to Shree DR.MDD

EAST = 1
NORTH = 1j
WEST = -1
SOUTH = -1j

class Robot:
    def __init__(self, orientation: complex, pos_x: int, pos_y: int):
        self.direction = orientation
        self._loc = pos_x + pos_y * 1j

    @property
    def coordinates(self) -> tuple[int, int]:
        return (int(self._loc.real), int(self._loc.imag))

    def move(self, path: str) -> None:
        for cmd in path:
            if cmd not in "RLA":
                raise ValueError(f"invalid instruction {cmd}")
            if cmd == "A":
                self._loc += self.direction
            elif cmd == "R":
                self.direction *= -1j
            elif cmd == "L":
                self.direction *= 1j
